node-js-client/lib/client.js

```diff
@@ -0,0 +1,140 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var request = require('request')
+var hawk = require('hawk')
+
+var generateHawkHeader = function (credentials, requestOptions) {
+  var header = hawk.client.header(requestOptions.uri, requestOptions.method, {
+    credentials: credentials,
+    ext: '',
+    contentType: 'application/json',
+    payload: requestOptions.body ? requestOptions.body : ''
+  })
+
+  return header
+}
+
+/**
+ * @class IPReputationClient
+ * @constructor
+ * @param {Object} config
+ *   @param {String} config.host
+ *   @param {Number} config.port
+ *   @param {id} config.id id for the HAWK header
+ *   @param {id} config.key key for the HAWK header
+ * @return {IPReputationServiceClient}
+ */
+var client = function(config) {
+  if (!Object.prototype.hasOwnProperty.call(config, 'host')) {
+    throw new Error('Missing required param host for IP Reputation Client.')
+  } else if (!Object.prototype.hasOwnProperty.call(config, 'port')) {
+    throw 'Missing required param port for IP Reputation Client.'
```

Comment:
Was this going to be used only server-side? If so, could possibly use [joi](https://github.com/hapijs/joi) to do object validation.